src/app/logic/quiz.js

```diff
@@ -0,0 +1,109 @@
+const axios = require('axios');
+
+const gameInfo = {};
+
+const fetchItems = (api) => {
+  axios
+    .get(api)
+    .then((item) => item.data)
+    .then((item) => {
+      gameInfo.questionScope.push(...item.results);
+      if (item.next !== null) {
+        fetchItems(item.next);
+      }
+      if (item.count === gameInfo.questionScope.length) {
+        gameInfo.questionScope = gameInfo.questionScope.map(
+          (value) => ({
+            name: value.name,
+            idx: value.url.split('/')[5],
+          }),
+        );
+        localStorage.setItem(
+          gameInfo.mode,
+          JSON.stringify(gameInfo.questionScope),
+        );
+      }
+    })
+    .catch((err) => {
+      throw Error(`fetchError: ${err}`);
+    });
+};
+
+const fetchQuestionScope = () => {
+  gameInfo.questionScope = [];
```

Comment:
avoid unnecessary mutations, set this array prop and prop type as initial value

Reply:
fetchQuestionScope is called only, when we don't have data for this mode in localStorage. If we have questionScope filled after one mode, and change to other without data in localStorage, new data will be added to the back of questionScope. This can lead to questions from other mode during play